submissions/Ant-C-tech/tiny-js-world/index.js

```diff
@@ -0,0 +1,97 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Ant-C-tech/a-tiny-JS-world
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+const dog = {
+   species: 'dog',
+   name: 'Spike',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: 'Woof-Woof!',
+   friends: ['John'],
+};
+
+const cat = {
+   species: 'cat',
+   name: 'Tom',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: 'Meow-Meow!',
+   friends: ['John'],
+};
+
+const woman = {
+   species: 'human',
+   name: 'Jane',
+   gender: 'female',
+   legs: 2,
+   hands: 2,
+   saying: 'Nice to meet you!',
+};
+
+const man = {
+   friends: ['Spike', 'Tom'],
+   species: 'human',
+   name: 'John',
+   legs: 2,
+   hands: 2,
+   saying: 'Hi, how are you?',
+   gender: 'male',
+};
+
+function createGreetingContent(obj) {
+
+   const propPrefixes = {
+      species: 'I am a %. ',
+      name: 'My name is %. ',
+      gender: 'My gender is %. ',
+      hands: 'I have % hands. ',
+      legs: 'I have % legs. ',
+      saying: 'I want to say you "%". ',
+      friends: 'I have friend %. ',
+   };
+
+  const greeting = Object.keys(propPrefixes).map((key) => {
+      if (Array.isArray(obj[key])) {
+         for (const item of obj[key]) {
+            return propPrefixes[key].replace('%', item);
+         };
+      } else if (obj[key] && propPrefixes[key]) {
+         return propPrefixes[key].replace('%', obj[key]);
+      };
+   });
```

Comment:
> Order of keys in an object is not guaranteed. Whenever a particualr order is required avoid Object.values|keys|entries methods as well as for ... in or for ... of.
List propertis to publish explicitly.

Please feel free asking a question if the above is not really clear.

Now that you have templates did you try treating `friends` just like any other property.

Reply:
Thank you for your review and for the opportunity to ask questions! I think you mean something like this:
const greeting = Object.keys(propPrefixes).map((key) => {
      if (obj[key] && propPrefixes[key]) {
         return propPrefixes[key].replace('%', obj[key]);
      };
   });
But now, I still need one condition to show correctly how many friends the character has: "1 friend" or  "many friendS". Please, help me  resolve this problem properly.

Reply:
Thank you for your review! I tried to do this way, but now when I treating "friends" just like any other property I have a problem with plural of word "friend". I think I need one more condition, but I am not sure you would think it as a good idea.